src/Plugin/Condition/DataIsEmpty.php

```diff
@@ -36,17 +36,10 @@ class DataIsEmpty extends RulesConditionBase {
   /**
    * {@inheritdoc}
    */
-  public function evaluate() {
-    $data = $this->getContext('data')->getContextData();
+  protected function doEvaluate($data) {
     if ($data instanceof ComplexDataInterface || $data instanceof ListInterface) {
       return $data->isEmpty();
```

Comment:
@klausi Not exactly.  The problem is that most implementations of ComplexDataInterface are configured to auto-unwrap; see the implementation of TypedDataManager::getCanonicalRepresentation(), which checks for a 'unwrap_for_canonical_representation' flag in a typed data definition. All common types I've looked at have this definition property TRUE.  So typically, the data unwraps.

Implementations of ContentEntity, however, do not unwrap, but there are places in the code that guarantee that the isEmpty() will always be false.  Entities want pretty badly to be "not empty".

So to get to that line of code, you need a complex data instance that (1) does not unwrap, and (2) can actually return isEmpty() as TRUE.  As you can see, you need to do a weird mock to get that.  I'm not sure of a practical use case for the combination, but the interfaces definitely allow for this.  So the code is not dead. But it is a weird edge case.